pkg/apis/core/validation/validation.go

```diff
@@ -3458,7 +3458,7 @@ func ValidatePodTemplate(pod *core.PodTemplate) field.ErrorList {
 // ValidatePodTemplateUpdate tests to see if the update is legal for an end user to make. newPod is updated with fields
 // that cannot be changed.
 func ValidatePodTemplateUpdate(newPod, oldPod *core.PodTemplate) field.ErrorList {
-	allErrs := ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta, field.NewPath("metadata"))
+	allErrs := ValidateObjectMetaUpdate(&newPod.ObjectMeta, &oldPod.ObjectMeta, field.NewPath("metadata"))
```

Comment:
@caesarxuchao this is the bug that caused pod template test failing.
There was a wrong arguments order, see https://github.com/kubernetes/kubernetes/blob/07240b7166d83bed49d783e0ecdfa7ee7e62cfca/pkg/apis/core/validation/validation.go#L341-L343

Reply:
good catch. that'll need backporting

Reply:
sure!

Reply:
raised a PR #60683

Reply:
Good catch! The bug prevented the garbage collector from removing finalizers because the validation code thought gc was trying to adding finalizers.